lib/utils.pm

```diff
@@ -2349,4 +2351,89 @@ sub package_upgrade_check {
     }
 }
 
+=head2 _validate_result
+    _validate_result();
+
+This is a private method which is used by C<generate_results> to convert the
+results in a string representation. At the moment the status that are supported
+are {PASS,FAIL}.
+
+The method takes as the only argument the return of a perl statement or
+subroutine.
+
+=cut
+
+sub _validate_result {
+    my $result = shift;
+    if ($result == 0) {
+        return 'PASS';
+    } elsif ($result == 1) {
+        return 'FAIL';
+    } else {
+        return undef;
+    }
+}
+
+=head2 generate_results
+    generate_results();
+
+This function is used to construct a hash suitable for representation in junit
+xml format.
+
+=cut
+
+sub generate_results {
+    my ($name, $description, $result) = @_;
+
+    my %results = (
+        test => $name,
+        description => $description,
+        result => _validate_result($result)
+    );
+    return %results;
+}
+
+=head2 pars_results
+    pars_results();
+
+Takes C<test> as an argument. C<test> is an array of hashes which contain the
+test results. They usually those are generated by C<generate_results>. Those
+are parsed and create the junit xml representation.
+
+=cut
+
+sub pars_results {
+    my ($testsuite, $xmlfile, @test) = @_;
+
+    # check if there are some single test failing
+    # and if so, make sure the whole testsuite will fail
+    my $fail_check = 0;
+    for my $i (@test) {
+        record_info "i", $i;
+        if ($i->{result} eq 'FAIL') {
+            $fail_check++;
+        }
+    }
+
+    if ($fail_check > 0) {
+        script_run(qq{echo "<testsuite name='$testsuite' errors='1'>" >> $xmlfile});
+    } else {
+        script_run(qq{echo "<testsuite name='$testsuite'>" >> $xmlfile});
+    }
+
+    # pars all results and provide expected xml file
```

Comment:
nit: parse

I know it's nothing, but may be we should use `parse` everywhere in here.

Reply:
*ready for review* edited only this one. But i think you are right. But let me do it in a separate PR as this would have to touch likely other parts